webui/module/Director/view/director/director/messages.phtml

```diff
@@ -63,7 +63,7 @@ $this->headTitle($title);
 
 function getDirectorMessages() {
    $.ajax({
-      url : '<?php echo $this->url('director', array('action' => 'getData'), null) . '?data=messages&limit=500&offset=null&reverse=true'; ?>',
+      url : '<?php echo $this->url('api/director', array(), array()) . '?messages=true&limit=500&offset=null&reverse=true'; ?>',
```

Comment:
Is the messages parameter required? How about `api/director/messages` ?

Reply:
From my point of view, this should be done with an additional child route and additional controller if you want `api/director/messages` as an endpoint. I'd like to see this changed in a separate PR afterwards to be honest.